extensions/resource-deployment/src/ui/deploymentDialog.ts

```diff
@@ -132,4 +195,25 @@ export class DeploymentDialog extends DialogBase {
 		this.notebookService.launchNotebook(this.deploymentProvider.notebook);
 		this.dispose();
 	}
+
+	private isValidSQLPassword(field: DialogFieldInfo, component: azdata.InputBoxComponent): boolean {
+		const password = component.value!;
+		// Validate SQL Server password
+		const userName = field.userName ? field.userName! : this.variables[field.userNameVariableName!];
+		const containsUserName = userName && password.includes(userName);
+		// Instead of using one RegEx, I am seperating it to make it more readable.
+		const hasUpperCase = /[A-Z]/.test(password) ? 1 : 0;
+		const hasLowerCase = /[a-z]/.test(password) ? 1 : 0;
+		const hasNumbers = /\d/.test(password) ? 1 : 0;
+		const hasNonalphas = /\W/.test(password) ? 1 : 0;
+		return !containsUserName && password.length >= 8 && password.length <= 128 && (hasUpperCase + hasLowerCase + hasNumbers + hasNonalphas >= 3);
```

Comment:
In a completely ideal world, we’d be able to tell customers which of these checks fails. Not going to block on that, but as a user, it could be convenient.